data/src/message.rs

```diff
@@ -1061,11 +1061,16 @@ fn monitored_targets_text(targets: Vec<String>) -> Option<String> {
 }
 
 pub fn reference_user(sender: NickRef, own_nick: NickRef, message: &Message) -> bool {
+    let contains_nick = |text: &str, nick: &str| {
+        text.split(|c: char| !c.is_alphanumeric())
+            .any(|word| word == nick)
+    };
```

Comment:
We should use the same logic as when we parse fragments, which is non whitespace and then compare word and trimmed word, both normal and lowercase with incoming nick